src/test/kotlin/com/exactpro/th2/common/message/TestMessageFilterUtils.kt

```diff
@@ -84,6 +103,32 @@ class TestMessageFilterUtils {
         Assertions.assertEquals(rootMessageFilterJson, objectMapper.writeValueAsString(toTreeTable))
     }
 
+    @ParameterizedTest
+    @MethodSource("additionalMetadata")
+    fun `valid root message filter to readable body collection conversion`(additionalMetadata: Map<String, String>) {
+        val expectedJson =
+            readableRootMessageFilterJson.replace(ADDITIONAL_METADATA_TAG, mapToJsonConverter(additionalMetadata))
+        val toTreeTable = RootMessageFilter.newBuilder().apply {
+            messageType = "MsgType"
+            messageFilter = createMessageFilter()
+            metadataFilterBuilder.apply {
+                putPropertyFilters("propA", simplePropertyFilter("valA", NOT_EQUAL, true))
+                putPropertyFilters("propB", simplePropertyFilter("valB"))
+            }
+            comparisonSettingsBuilder.apply {
+                addIgnoreFields("fieldA")
+                addIgnoreFields("fieldB")
+            }
+        }.build().toReadableBodyCollection(additionalMetadata)
+
+        val expected = objectMapper.readTree(expectedJson).toList()
+        val actual = objectMapper.valueToTree<JsonNode>(toTreeTable).toList()
```

Comment:
Why can we simply compare two JsonNode? Why do we need to call `toList()`

Reply:
Our JsonNode is a list of serialized TreeTable. Two list elements are only compared as equal if they have the same values in the exact same order. In our case, the expected list has `Metadata` as the last element in the list and in the actual list as the first element. So, to compare two unordered lists, we have to convert the JsonNode to a list and check their size and elements

Reply:
If we just compare two JsonNodes, we get failed check because the elements are unordered